runtime/src/iree/io/parameter_provider.h

```diff
@@ -0,0 +1,241 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_IO_PARAMETER_PROVIDER_H_
+#define IREE_IO_PARAMETER_PROVIDER_H_
+
+#include <stdint.h>
+
+#include "iree/base/api.h"
+#include "iree/hal/api.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+//===----------------------------------------------------------------------===//
+// iree_io_parameter_provider_t
+//===----------------------------------------------------------------------===//
+
+// Indicates an event signaled from the hosting program.
+typedef enum iree_io_parameter_provider_signal_e {
+  // Program is resuming from a suspended state.
+  // Providers may reallocate memory for pools and caches.
+  IREE_IO_PARAMETER_PROVIDER_SIGNAL_RESUME = 0,
+  // Program is entering a suspended state.
+  // Providers should drop any transient memory that is possible to reallocate
+  // upon resume.
+  IREE_IO_PARAMETER_PROVIDER_SIGNAL_SUSPEND = 1,
+  // Program has received a low memory alert.
+  // Providers must aggressively drop all possible memory even if expensive to
+  // rematerialize it. On some platforms this is sent as a threat that if
+  // sufficient memory is not unwired/freed ASAP the process will be killed.
+  IREE_IO_PARAMETER_PROVIDER_SIGNAL_LOW_MEMORY = 2,
```

Comment:
And on some platforms, the process will be killed anyway 🙃

Reply:
/me eyes ios